platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/migratable/FeatureProcessor.java

```diff
@@ -179,135 +300,118 @@ public boolean stopFeature(ProfileMigrationReport report, Feature feature, Strin
    * Processes features by recording tasks to start, stop, install, or uninstall features that were
    * originally in the corresponding state.
    *
-   * @param report the report where to record errors
    * @param jprofile the profile where to retrieve the set of features from the original system
    * @param tasks the task list where to record tasks to be executed
-   * @return <code>false</code> if we failed to retrieve from memory a left over feature; <code>
-   *     true</code> if everything was ok
    */
-  public boolean processFeatures(
-      ProfileMigrationReport report, JsonProfile jprofile, TaskList tasks) {
-    LOGGER.debug("Processing features import");
-    final Map<String, JsonFeature> jfeaturesMap =
-        jprofile
-            .features()
-            .collect( // linked map to preserve order
-                org.codice.ddf.admin.application.service.migratable.Collectors.toLinkedMap(
-                    JsonFeature::getId, Function.identity()));
-
-    processMemoryFeatures(jfeaturesMap, tasks);
-    return processLeftoverExportedFeatures(report, jfeaturesMap, tasks);
+  public void processFeaturesToRecordTasks(JsonProfile jprofile, TaskList tasks) {
+    LOGGER.debug("Processing features");
+    final Map<String, Feature> featuresMap =
+        Stream.of(listFeatures("Import"))
+            .collect(Collectors.toMap(Feature::getId, Function.identity()));
+
+    processExportedFeaturesToRecordTasks(jprofile, featuresMap, tasks);
+    processLeftoverFeaturesToRecordTasks(featuresMap, tasks);
   }
 
   /**
-   * Processes features in memory by recording tasks to start, stop, install, or uninstall features
+   * Processes exported features by recording tasks to start, stop, install, or uninstall features
    * that were originally in the corresponding state.
    *
    * <p><i>Note:</i> Any features found in memory are removed from the provided map since they have
    * been processed.
    *
-   * @param jfeatures the set of features on the original system
+   * @param jprofile the profile where to retrieve the set of features from the original system
+   * @param features the set of features in memory
    * @param tasks the task list where to record tasks to be executed
    */
-  public void processMemoryFeatures(Map<String, JsonFeature> jfeatures, TaskList tasks) {
-    LOGGER.debug("Processing features defined in memory");
-    final Feature[] features = listFeatures("Import");
-
-    for (final Feature feature : features) {
-      final JsonFeature jfeature =
-          jfeatures.remove(feature.getId()); // remove from jfeatures when we find it
-
-      processFeature(jfeature, feature, tasks);
-    }
+  public void processExportedFeaturesToRecordTasks(
+      JsonProfile jprofile, Map<String, Feature> features, TaskList tasks) {
```

Comment:
❓ Same question as with the `BundleProccessor`, would it be worth creating a constructor and having the `FeatureMigrator` pass in these arguments so they don't have to be used as arguments everywhere?

Reply:
The first pass at this had all the functionality of the migrators as part of the processors themselves. That did tremendously complicate testing which is why it was moved to a separate utility class. I guess at this point I saw that as an extension to the processor which is why I created it at init time.

With the changed requested to move the initialization to blueprint, I am not sure I want to undo everything again to redo it how it was done before.